actions/add-review-url/src/AddReviewUrl.js

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+const {graphql} = require('@octokit/graphql');
+const core = require('@actions/core');
+
+class AddComment {
+
+  constructor(comment)
+  {
+    this.addComment(comment);
+  }
+
+  getOwnerAndRepo() {
+    if (process.env.GITHUB_REPOSITORY) {
+      const [owner, repo] = process.env.GITHUB_REPOSITORY.split('/');
+      return {'owner': owner, 'repo': repo};
+
+    } else {
+      throw new Error('not able to obtain GITHUB_REPOSITORY from process.env');
+    }
+  }
+
+  getGraphqlWithAuth() {
+    const token = core.getInput('repo-token');
+    return graphql.defaults({
+      headers: {
+        authorization: `token ${token}`,
+      },
+    });
+  }
+
+  addPullRequestCommentMutation() {
+    return `mutation AddPullRequestComment($subjectId: ID!, $body: String!) {
+  addComment(input:{subjectId:$subjectId, body: $body}) {
+    commentEdge {
+        node {
+        createdAt
+        body
+      }
+    }
+    subject {
+      id
+    }
+  }
+}`;
+  }  
+
+  getPullNumber() {
+    if (process.env.GITHUB_REF) {
+      return parseInt(process.env.GITHUB_REF.split('/')[2]);
+    }
+  }
+
+  getGraphqlWithAuth() {

```

Comment:
Is this duplicated?

Reply:
Good catch. It is interesting that this was not flagged as compilation error. Updated.